src/fileio.cpp

```diff
@@ -977,55 +973,67 @@ bool DoScanWorkingDirectory()
 	return _searchpaths[SP_PERSONAL_DIR] != tmp;
 }
 
+/**
+ * Gets the home directory of the user.
+ * May return an empty string in the unlikely scenario that the home directory cannot be found.
+ * @return User's home directory
+ */
+static std::string GetHomeDir()
+{
+	std::string homedir;
+#ifdef __HAIKU__
+	BPath path;
+	find_directory(B_USER_SETTINGS_DIRECTORY, &path);
+	homedir = path.Path();
+#else
+	char *home_env = getenv("HOME");
+	if (home_env != nullptr) {
+		homedir = home_env;
+	} else  {
```

Comment:
too many spaces, but I would also rewrite this function a bit, as this is really hard to read.

Why not return in the `if` cases? Something like:

```
if (home_env != nullptr) {
  ..
  return homedir;
}

pw = getpwuid()
if (pw != nullptr) return pw->pw_dir
```

Might make it a lot easier to read, instead of this nesting?